lib/dry/system/booter.rb

```diff
@@ -16,29 +16,38 @@ module System
     #
     # @api private
     class Booter
-      attr_reader :path
+      attr_reader :paths
 
       attr_reader :booted
 
       attr_reader :components
 
       # @api private
-      def initialize(path)
-        @path = path
+      def initialize(paths)
+        @paths = paths
         @booted = []
         @components = ComponentRegistry.new
       end
 
       # @api private
       def bootable?(component)
-        boot_file(component).exist?
+        boot_file(component)&.exist? # FIXME maybe don't need exist
       end
 
       # @api private
       def boot_file(name)
         name = name.respond_to?(:root_key) ? name.root_key.to_s : name
 
-        path.join("#{name}#{RB_EXT}")
+        # path.join("#{name}#{RB_EXT}")
+
+        find_boot_file(name)
+      end
+
+      def find_boot_file(name)
+        paths.detect { |path|
+          file_name = path.join("#{name}#{RB_EXT}")
+          break file_name if File.exist?(file_name)
+        }
```

Comment:
@timriley this should raise an exception if it finds > 1 file

Reply:
@solnic this is actually expressly designed to support different boot dirs having files with the same name. The idea is that this is like a shell $PATH, so you can prepend your own bootable first to provide a customized replacement of a bootable component that you’d find within a later entry in the dirs. 

See the attached Hanami PR for a demonstration of why I want it to be that way — Hanami can provide its own standard boot files, but still allow app authors to swap them out for their own implementations on a fine-grained basis.